From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Tue, 21 Mar 2023 15:24:33 +0100
Subject: [PATCH] Don't apply de-synced opencount for shulker boxes

Would bug out when opening a shulker box multiple times in a tick, caused by ping or fast clicking.

The issue is caused by how block events in minecraft work. Just commenting this line out prevents
the wrong data from being set - there isn't a reason to do this on the server anyway.

diff --git a/src/main/java/net/minecraft/world/level/block/entity/ShulkerBoxBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/ShulkerBoxBlockEntity.java
index 1fa22445a4ecc8c08dbcf0cc6bd39dc5003604c4..89c0fbe124cdb9a2f2ed2253eece86c755504f1d 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/ShulkerBoxBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/ShulkerBoxBlockEntity.java
@@ -179,7 +179,7 @@ public class ShulkerBoxBlockEntity extends RandomizableContainerBlockEntity impl
     @Override
     public boolean triggerEvent(int type, int data) {
         if (type == 1) {
-            this.openCount = data;
+            // this.openCount = data; // CloudPlane
             if (data == 0) {
                 this.animationStatus = ShulkerBoxBlockEntity.AnimationStatus.CLOSING;
             }
